### ISSUE TITLE
Plugins using setuptools entry points resolve to system and not setup_requires

### ISSUE THREAD

**Issue by anntzer** [pypa/setuptools#2158]
Assume that you have e.g. setuptools_scm 3 installed system-wide, and try to to run the following setup.py:
```python
from setuptools import setup
setup(name="foo", setup_requires=["setuptools_scm>=4"])
```
you get the warning
```
/usr/lib/python3.8/site-packages/setuptools/dist.py:701: UserWarning: Module setuptools_scm was already imported from /usr/lib/python3.8/site-packages/setuptools_scm/__init__.py, but /tmp/foo/.eggs/setuptools_scm-4.1.1-py3.8.egg is being added to sys.path
```
(i.e. the system-wide setuptools_scm was loaded, even though it is too old).  One can confirm that the "correct" newer setuptools_scm indeed doesn't get used by changing setup.py to
```python
from setuptools import setup
setup(name="foo",
      setup_requires=["setuptools_scm>=4"],
      use_scm_version={"parentdir_prefix_version": "foo-"})
```
in which case one gets the same warning and then a traceback
```
TypeError: __init__() got an unexpected keyword argument 'parentdir_prefix_version'
```

I guess a solution may be e.g. to move the entry point (which is what gets eagerly loaded by setuptools) to a separate package (let's say `_setuptools_scm_helper`) which itself imports setuptools_scm only when actually needed with a pass-through shim so that setuptools only imports `_setuptools_scm_helper` and then the correct version of setuptools_scm can be loaded when needed.

**Comment by jaraco** [comment]
I've transferred the issue to the setuptools project, as setuptools_scm is just one plugin affected.

**Comment by jaraco** [comment]
Released as v44.1.1 and v47.1.1.

**Comment by anntzer** [comment]
Thanks for the quick fix! :)
